pkg/runtime/ref/injector.go

```diff
@@ -38,15 +38,22 @@ func (i *DefaultInjector) InjectSecretRef(items []*ref.Item, metaData map[string
 
 	meta := make(map[string]string)
 	for _, item := range items {
-		store := i.Container.GetSecretStore(item.Name)
+		store := i.Container.GetSecretStore(item.StoreName)
 		secret, err := store.GetSecret(secretstores.GetSecretRequest{
 			Name: item.Key,
 		})
 		if err != nil {
 			return metaData, err
 		}
 		for k, v := range secret.Data {
-			meta[k] = v
+			if k != item.SubKey {
```

Comment:
What happened if the component returns only one secret instead of a secret list?

Reply:
this condition, the user also should use sub_key, mean key = sub_key